### PR TITLE
Remove backend legacy alerting code

### LINKED ISSUE #12988: Remove API code for legacy alerting management pages.
## Description

A while ago we rewrote the alerting UI. With [this PR](https://github.com/Graylog2/graylog2-server/pull/12987) we removed the related legacy UI code. The next step is removing the related backend code.

Here is an overview of the no longer needed API routes.

```
/streams/${streamId}/alerts/conditions
/streams/${streamId}/alerts/conditions/${alertConditionId}
/streams/${streamId}/alerts/sendDummyAlert
/streams/${streamId}/alerts/conditions/${conditionId}/test
/alerts/conditions/types
/alerts/conditions
/streams/alerts/${alertId}
/streams/${streamId}/alerts
/streams/alerts/paginated
/streams/alerts
```

Please note we still want to support legacy alarm callbacks, until the next major release.

## Your Environment

* Graylog Version: 4.3

### PR DESCRIPTION
**Pull request by patrickmann**
Resolves #12988 

Removes backend code that relates to stream-based alerting.

## How tested
- All tests pass
- Created HTTP alert, script alert, legacy script alert on master
- Switched to this branch
- Verified that
  - No DB errors due to changed schemas
  - Still able to create / edit / view alerts and legacy alerts
  - Both script and legacy script alerts are triggered
  - Created and installed a content pack containing legacy alarms

### CODE REVIEW

**Review by mpfz0r — COMMENTED** [review]
I left some comments.

We could mark a few things as `@Depreacted` e.g. the alerting stuff in `StreamDTO` 

Also,we should have a comment in `UPGRADING.md` about this.
e.g. that old alerts from content packs will simply be ignored from now on.

File: graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStats.java [hunk 1]
```diff
@@ -88,8 +85,7 @@ public static ClusterStats create(ElasticsearchStats elasticsearchStats,
                                       long globalInputCount,
                                       Map<String, Long> inputCountByType,
                                       long extractorCount,
-                                      Map<Extractor.Type, Long> extractorCountByType,
-                                      AlarmStats alarmStats) {
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStats.java:92]: The `AlarmStats` class can also be deleted

**Review by mpfz0r — CHANGES_REQUESTED** [review]
Also `Stream` and `StreamImpl` has still `getAlertReceivers()` which can go away

File: graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java [hunk 2]
```diff
@@ -1,106 +0,0 @@
...
-    }
-
-    @Override
-    public boolean leaderOnly() {
-        return true;
-    }
-
-    @Override
-    public boolean startOnThisNode() {
-        return configuration.isEnableLegacyAlerts();
```
[the remark this answers is not in the document]
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java:94]: The entire AlertScannerThread class is deleted.

[the remark this answers is not in the document]
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java:94]: I've meant to delete `configuration.isEnableLegacyAlerts()`

[the remark this answers is not in the document]
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java:94]: Got it - removed that

[the remark this answers is not in the document]
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java:94]: there's more:
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/Configuration.java#L149-L151

[the remark this answers is not in the document]
Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java:94]: @patrickmann ping ^^ :)

[the remark this answers is not in the document]
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/periodical/AlertScannerThread.java:94]: Good catch

**Review by mpfz0r — APPROVED** [review]
LGTM :+1:

**Pull request merged**